dd-sdk-android/src/test/kotlin/com/datadog/android/rum/resource/RumResourceInputStreamTest.kt

```diff
@@ -0,0 +1,649 @@
+import com.datadog.android.rum.RumMonitor
+import com.datadog.android.rum.RumResourceKind
+import com.nhaarman.mockitokotlin2.doAnswer
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.doThrow
+import com.nhaarman.mockitokotlin2.inOrder
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.annotation.BoolForgery
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.annotation.LongForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStream
+import java.lang.RuntimeException
+import kotlin.math.max
+import kotlin.math.min
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Assume.assumeTrue
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class RumResourceInputStreamTest {
+
+    lateinit var testedInputStream: RumResourceInputStream
+
+    @Mock
+    lateinit var mockRumMonitor: RumMonitor
+
+    @Mock
+    lateinit var mockInputStream: InputStream
+
+    @StringForgery
+    lateinit var fakeUrl: String
+
+    @StringForgery
+    lateinit var fakeMessage: String
+
+    @BeforeEach
+    fun `set up`() {
+        GlobalRum.registerIfAbsent(mockRumMonitor)
+
+        testedInputStream = RumResourceInputStream(mockInputStream, fakeUrl)
+
+        // 𝕄 start resource 𝕎 init
+        verify(mockRumMonitor).startResource(
+            testedInputStream.key,
+            RumResourceInputStream.METHOD,
+            fakeUrl,
+            emptyMap()
+        )
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        GlobalRum.isRegistered.set(false)
+        GlobalRum.monitor = NoOpRumMonitor()
+    }
+
+    // region Atomic Methods
+
+    @Test
+    fun `𝕄 return byte 𝕎 read()`(
+        @IntForgery(-128, 127) byte: Int
+    ) {
+        // Given
+        whenever(mockInputStream.read()) doReturn byte
+
+        // When
+        val result = testedInputStream.read()
+
+        // Then
+        assertThat(result).isEqualTo(byte)
+        verifyNoMoreInteractions(mockRumMonitor)
+    }
+
+    @Test
+    fun `𝕄 fill byte array 𝕎 read(array)`(
+        @StringForgery text: String
+    ) {
+        // Given
+        val input = text.toByteArray()
+        val length = min(8, input.size)
```

Comment:
why using a min here ?

Reply:
To add variations, I just want to read part of the input stream (just 8 bytes), or the whole stream (if it's less than 8 bytes long).